### DISCUSSION TITLE
ReadPoint parameter stuck with value y15455 (Warning)

### DISCUSSION THREAD

**Discussion by rhermsen** [msillano/tuyaDAEMON#2] [General]
Hi Marco,

Thanks for sharing the [PM Detector](https://github.com/msillano/tuyaDAEMON/wiki/custom-device-'PM-detector':-case-study) case study details.
I also got one of these devices and did capture the communication details between PM25 application and PM Detector, and found your site while googling using some of the captured parameter names.
(I used Wireshark with [USBpcap](https://github.com/desowin/usbpcap/) for this)

Just to add a note, while 'playing' with the interface I managed to get the WritePoint and ReadPoint parameters 
 in an undefined state:
"WritePoint":"\xf567295","ReadPoint":"\xf567295"
The WritePoint value got cleared after a clear of the internal memory (e.g. {"fun":"06"}).
And at some point the ReadPoint value changed to 'ReadPoint': 'y15455', but seems to be stuck at that value.
I suspect this issue was triggered by sending a less than 3 digit 'storeTime' string e.g. "01" or "02" and leave the device running for a day or two.
(In python I now have: initStoreTimeString = '{"fun":"02","storetime":"' + storeTime.zfill(3) + '"}\n')

Btw, here is another interesting link I found for this PM Detector:
https://frdmtoplay.com/patching-in-fahrenheit/

**Comment by msillano** [comment]
Thanks for useful info and link.
I will try to reproduce this problem.

Best regard.
m.s.
